### ISSUE TITLE
Inconsistent generated typing for blocks (missing id and blockName)

### ISSUE THREAD

**Issue by DrMint** [payloadcms/payload#5216]
### Link to reproduction

:(

### Describe the Bug

When using blocks in different situations (within a blockfield, within the lexical rich text editor), the typing is extremely similar across the board, with the exception of: a block within a blockfield, within a block, within Lexical BlocksFeature.

In this specific case, the id and blockName properties are missing from the generated type for this block.

Found:
```ts
export interface BlockC {
  blockType: 'blockC';
}
```

Expected:
```ts
export interface BlockC {
  id?: string | null;
  blockName?: string | null;
  blockType: 'blockC';
}
```

## Collection

```ts
import { BlocksFeature, lexicalEditor } from "@payloadcms/richtext-lexical";
import { Block, CollectionConfig } from "payload/types";

const blockA: Block = {
  slug: "blockA",
  interfaceName: "BlockA",
  fields: [],
};

const blockB: Block = {
  slug: "blockB",
  interfaceName: "BlockB",
  fields: [],
};

const blockC: Block = {
  slug: "blockC",
  interfaceName: "BlockC",
  fields: [],
};

const blockD: Block = {
  slug: "blockD",
  interfaceName: "BlockD",
  fields: [{ name: "blocks", type: "blocks", blocks: [blockC] }],
};

const blockE: Block = {
  slug: "blockE",
  interfaceName: "BlockE",
  fields: [],
};

const blockF: Block = {
  slug: "blockF",
  interfaceName: "BlockF",
  fields: [{ name: "blocks", type: "blocks", blocks: [blockE] }],
};

export const Bugs: CollectionConfig = {
  slug: "bugs",
  fields: [
    { name: "blocks", type: "blocks", blocks: [blockA, blockF] },
    {
      name: "richText",
      type: "richText",
      editor: lexicalEditor({
        features: [BlocksFeature({ blocks: [blockB, blockD] })],
      }),
    },
  ],
};

```

So we have the following blocks:
- BlockA: simple block in a block field
- BlockB: simple block in a lexical rich text editor
- BlockC: simple block inside BlockD's block field
- BlockD: a block with a block field, in the rich text editor
- BlockE: simple block inside BlockF's block field
- BlockF: a block with a block field, in the collection's block field

## Generating the types
Run `npm run generate:types`

## Results

In the generated types, we got:
```ts
/**
 * This interface was referenced by `Config`'s JSON-Schema
 * via the `definition` "bugs".
 */
export interface Bug {
  id: string;
  blocks?: (BlockA | BlockF)[] | null;
  richText?: {
    root: {
      children: {
        type: string;
        version: number;
        [k: string]: unknown;
      }[];
      direction: ('ltr' | 'rtl') | null;
      format: 'left' | 'start' | 'center' | 'right' | 'end' | 'justify' | '';
      indent: number;
      type: string;
      version: number;
    };
    [k: string]: unknown;
  } | null;
  updatedAt: string;
  createdAt: string;
}
/**
 * This interface was referenced by `Config`'s JSON-Schema
 * via the `definition` "BlockA".
 */
export interface BlockA {
  id?: string | null;
  blockName?: string | null;
  blockType: 'blockA';
}
/**
 * This interface was referenced by `Config`'s JSON-Schema
 * via the `definition` "BlockB".
 */
export interface BlockB {
  id?: string | null;
  blockName?: string | null;
  blockType: 'blockB';
}
/**
 * This interface was referenced by `Config`'s JSON-Schema
 * via the `definition` "BlockC".
 */
export interface BlockC {
  blockType: 'blockC';
}
/**
 * This interface was referenced by `Config`'s JSON-Schema
 * via the `definition` "BlockD".
 */
export interface BlockD {
  blocks?: BlockC[] | null;
  id?: string | null;
  blockName?: string | null;
  blockType: 'blockD';
}
/**
 * This interface was referenced by `Config`'s JSON-Schema
 * via the `definition` "BlockE".
 */
export interface BlockE {
  id?: string | null;
  blockName?: string | null;
  blockType: 'blockE';
}
/**
 * This interface was referenced by `Config`'s JSON-Schema
 * via the `definition` "BlockF".
 */
export interface BlockF {
  blocks?: BlockE[] | null;
  id?: string | null;
  blockName?: string | null;
  blockType: 'blockF';
}

```

So we can see that all the blocks, with the exception of BlockC (the block within a block field, within Lexical rich text editor) has the id and blockName properties. Which seems to be a bug because BlockC definitely can have a BlockName:

![Screenshot_20240301_022956](https://github.com/payloadcms/payload/assets/29893320/71daa1f3-a8b9-4dd3-875b-1a8fb4275a0b)

And we can see it in the actual data when fetch from the API:

```json
{
  "id": "65e1181a6b264bdd76f3aa98",
  "blocks": [
    {
      "id": "65e1181a233a1c8e340f6356",
      "blockName": "A block name",
      "blockType": "blockA"
    },
    {
      "blocks": [
        {
          "id": "65e12f4919ec341100999ddb",
          "blockName": "A block name",
          "blockType": "blockE"
        }
      ],
      "id": "65e12f4819ec341100999dda",
      "blockName": "A block name",
      "blockType": "blockF"
    }
  ],
  "richText": {
    "root": {
      "type": "root",
      "format": "",
      "indent": 0,
      "version": 1,
      "children": [
        {
          "format": "",
          "type": "block",
          "version": 2,
          "fields": {
            "id": "65e1178f538c955798d01fe2",
            "blockName": "A block name",
            "blockType": "blockB"
          }
        },
        {
          "format": "",
          "type": "block",
          "version": 2,
          "fields": {
            "id": "65e12f3e19ec341100999dd8",
            "blockName": "A block name",
            "blockType": "blockD",
            "blocks": [
              {
                "id": "65e12f4019ec341100999dd9",
                "blockType": "blockC",
                "blockName": "A block name"
              }
            ]
          }
        }
      ],
      "direction": null
    }
  },
  "createdAt": "2024-02-29T23:49:46.352Z",
  "updatedAt": "2024-03-01T01:31:11.621Z"
}
```

### Payload Version

2.11.2

### Adapters and Plugins

"@payloadcms/bundler-webpack": "1.0.6",     "@payloadcms/db-mongodb": "1.4.3",     "@payloadcms/richtext-lexical": "0.7.0",

**Comment by remy90** [comment]
@DrMint I can see your bug specifically relates to typing but have you encountered any runtime errors relating to `blockName` not being defined on blocks?  I have, despite not type errors and definitions for `name`, where applicable.

**Comment by DrMint** [comment]
Hi @remy90, I have included the response when fetching the document: the blockName and id is present in all those situation. So no runtime errors on my side.

**Comment by AlessioGr** [comment]
Hey @DrMint ! Good news, this has been fixed in the latest v3 beta of richtext-lexical. Do note that the fix won't be backported to 2.0, so I recommend upgrading.

The issue was causes by inconsistent lexical sub-field sanitization behavior, which sometimes omitted blockName / blockType from sub blocks. Sub-field sanitization has been completely rewritten in v3. Not only does it run a lot less often and is faster now, it's now sanitizing any amount of sub-fields correctly!

If you encounter something similar again in 3.0, please feel free to open a new issue :)
